src/config.rs

```diff
@@ -218,28 +218,44 @@ pub fn load_config(config_path: &PathBuf, password: &ZeroingString) -> Result<Se
 	})
 }
 
-pub fn get_grin_path() -> PathBuf {
+pub fn get_grin_path(chain_type: &ChainTypes) -> PathBuf {
 	let mut grin_path = match dirs::home_dir() {
 		Some(p) => p,
 		None => PathBuf::new(),
 	};
 	grin_path.push(GRIN_HOME);
-	grin_path.push(CHAIN_NAME);
+	grin_path.push(chain_type.shortname());
 	grin_path
 }
 
-pub fn node_secret_path() -> PathBuf {
-	let mut path = get_grin_path();
+pub fn node_secret_path(chain_type: &ChainTypes) -> PathBuf {
+	let mut path = get_grin_path(chain_type);
 	path.push(NODE_API_SECRET_FILE_NAME);
 	path
 }
 
-pub fn wallet_owner_secret_path() -> PathBuf {
-	let mut path = get_grin_path();
+pub fn wallet_owner_secret_path(chain_type: &ChainTypes) -> PathBuf {
+	let mut path = get_grin_path(chain_type);
 	path.push(WALLET_OWNER_API_SECRET_FILE_NAME);
 	path
 }
 
+pub fn grin_node_url(chain_type: &ChainTypes) -> SocketAddr {
+	if *chain_type == ChainTypes::Testnet {
+		"127.0.0.1:13413".parse().unwrap()
+	} else {
+		"127.0.0.1:3413".parse().unwrap()
+	}
+}
+
+pub fn wallet_owner_url(chain_type: &ChainTypes) -> SocketAddr {
+	if *chain_type == ChainTypes::Testnet {
+		"127.0.0.1:13420".parse().unwrap()
```

Comment:
If I recall correctly, the wallet port is 3420 by default also for testnet

Reply:
You're correct, the default grin-wallet.toml for testnet is generated with `owner_api_listen_port = 3420`. Do you know why we're using the same owner port as mainnet?

Reply:
My guess is that two different sets of people were doing node and wallet software and some discrepancies came to life. Maybe @yeastplume can tell more